### PR TITLE
ensure that chef-stable repository is disabled if present to prevent unintended updates

### PR DESCRIPTION
**Pull request by marcparadise**
Testing is locally in flight, but opening this up for review in the mean time. 

ping @chef/lob @mmzyk @sean-horn

### CODE REVIEW

**Comment by marcparadise** [comment]
force-pushed an amended commit to remove a change we don't need here, and fix a comment

**Comment by stevendanna** [comment]
:+1: Looks good

**Comment by marcparadise** [comment]
Our default `['addons']['packages']` attribute containing a list of all available packages, and the add-ons recipe just loops through and installs everything it sees in `packages`.  This means on reconfigure we'd auto-install all add-ons the first time we did a reconfigure with this recipe included. 

I've modified this PR to split out the addon repo setup into its own recipe and invoke that instead of the base add-on wrapper.

**Comment by marcparadise** [comment]
The failure here appears unrelated. Running it again and will take a look later if it continues to fail.

File: omnibus/files/private-chef-cookbooks/private-chef/recipes/add_ons_repository.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+#
+# All Rights Reserved
```
Comment (stevendanna) [inline omnibus/files/private-chef-cookbooks/private-chef/recipes/add_ons_repository.rb]: This isn't correct anymore, let's fix it.

**Comment by stevendanna** [comment]
:+1: Please fix the copyright notices if you get a chance.

**Comment by jeremiahsnapp** [comment]
I love it @marcparadise . Great fix.

**Comment by toioiz** [comment]
This breaks other items that use chef_ingredient and try to install more than 1 at a time.

Example:
Write chef-stable.repo (enabled)
chef_ingredient "manage"
  -- manage reconfigure
  -- chef-server-ctl reconfigure (disables)
chef_ingredient "reporting" (panic error)
chef_ingredient "push-jobs"

**Comment by marcparadise** [comment]
Well crap.   That seems to rule out any other possible paths to fix this, unless we explicitly enable the resource beforehand , and disable it after.  Ugh.

**Comment by toioiz** [comment]
You can call --enable-repo=chef-stable during the yum command, not sure if the yum resource will allow you to do that inside the chef_ingredient lwrp.

ie. yum --enable-repo=chef-stable install opscode-manage

Note: this may require changes to the yum_package lwrp as well :(

**Pull request merged**